src/Web/Controllers/AccountController.cs

```diff
@@ -73,14 +81,48 @@ public async Task<IActionResult> SignIn(LoginViewModel model, string returnUrl =
             }
             ViewData["ReturnUrl"] = returnUrl;
 
+            ApplicationUser applicationUser = new ApplicationUser
+            {
+                UserName = model.Email
+            };
+
+            var signIn = new SignIn
+            {
+                SignInId = Guid.NewGuid().ToString(),
+                PasswordHash = _userManager.PasswordHasher.HashPassword(applicationUser, model.Password),
+                MerchantLocalDate = DateTimeOffset.Now,
+                CustomerLocalDate = model.DeviceFingerPrinting.ClientDate,
+                UserId = model.Email,
+                DeviceContextId = model.DeviceFingerPrinting.SessionId,
+                AssessmentType = AssessmentType.Protect.ToString(),
+                CurrentIpAddress = _contextAccessor.HttpContext.Connection.RemoteIpAddress.ToString()
+            };
+
+            var signInResponse = await _fraudProtectionService.PostSignIn(signIn);
+
+            var fraudProtectionIO = new FraudProtectionIOModel(signIn, signInResponse, "SignIn");
+            TempData.Put(FraudProtectionIOModel.TempDataKey, fraudProtectionIO);
+
+
             var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, lockoutOnFailure: false);
```

Comment:
We shouldn't call this PasswordSignInAsync until after the random rejection, and even then, only in the not rejected case.

Reply:
Updated the order. Please check and let me know if the change looks good.